src/core/model_config.proto

```diff
@@ -623,6 +623,70 @@ message ModelOptimizationPolicy
   PinnedMemoryBuffer output_pinned_memory = 6;
 }
 
+//@@
+//@@.. cpp:var:: message ModelQueuePolicy
+//@@
+//@@   Timeout policy for inference requests. If specified, the scheduler will
+//@@   handle pending requests that exceed timeout accordingly.
+//@@
+message ModelQueuePolicy
+{
+  //@@
+  //@@  .. cpp:enum:: Action
+  //@@
+  //@@     The action applied on timeout request.
+  //@@
+  enum Action {
+    //@@    .. cpp:enumerator:: Action::REJECT = 0
+    //@@
+    //@@       Reject the request and return error message accordingly.
+    //@@
+    REJECT = 0;
+
+    //@@    .. cpp:enumerator:: Action::DELAY = 1
+    //@@
+    //@@       Delay the request until all other requests at the same
+    //@@       (or higher) priority levels that have not reached their timeouts
+    //@@       are processed. A delayed request will eventually be processed,
+    //@@       but may be delayed indefinitely due to newly arriving requests.
+    //@@
+    DELAY = 1;
+  }
+
+  //@@
+  //@@  .. cpp:var:: Action action
+  //@@
+  //@@     The action applied on timeout request. The default action is REJECT.
+  //@@
+  Action action = 1;
+
+  //@@
+  //@@  .. cpp:var:: uint64 default_timeout_microseconds
+  //@@
+  //@@     The default timeout for every request, in microseconds.
+  //@@     The default value is 0 which indicates that no timeout is set.
+  //@@
+  uint64 default_timeout_microseconds = 2;
+
+  //@@
+  //@@  .. cpp:var:: bool allow_timeout_override
+  //@@
+  //@@     Whether individual request can override the timeout value.
+  //@@     The default value is false.
+  //@@
+  bool allow_timeout_override = 3;
+
+  //@@
+  //@@  .. cpp:var:: uint32 max_queue_size
+  //@@
+  //@@     The maximum queue size for holding requests. The request enqueued
```

Comment:
"The maximum ... . A request will be rejected immediately if it can't be enqueued because the queue is full. The default value is 0 ..."

Right now we always reject. In the future we may want to add a policy setting for queue size as well. For example, we might want to allow the option to have the request enqueue in a lower-priority queue if its queue is full.